src/sdk/config_fs.js

```diff
@@ -265,104 +440,222 @@ class ConfigFS {
+        await native_fs_utils.create_config_file(this.fs_context, account_dir_path, account_path, data_string);
+        await this.link_account_name_index(_id, name, owner);
+        await this.link_access_keys_index(_id, account_data.access_keys);
     }
 
     /**
      * update_account_config_file updates account config file
      * if old_access_keys is an array that contains at least 1 item -
      * unlink old access_keys and link new access_keys
-     * @param {string} account_name
+     * 1. update /identities/account_id/identity.json
+     * 2. if name updated -
+     *    link /accounts_by_name/new_account_name -> /identities/account_id/identity.json
+     *    unlink /accounts_by_name/old_account_name
+     * 3. if access key was updated -
+     *    for all new_access_keys - link /access_keys/new_access_key -> /identities/account_id/identity.json
+     *    for all old_access_keys - unlink /access_keys/old_access_key
      * @param {Object} account_new_data
-     * @param {Object[]} [new_access_keys_to_link]
-     * @param {String[]} [old_access_keys_to_remove]
+     * @param {{old_name?: string, new_access_keys_to_link?: { access_key: string }[], access_keys_to_delete?: { access_key: string }[]}} [options]
      * @returns {Promise<void>}
      */
-    async update_account_config_file(account_name, account_new_data, new_access_keys_to_link = [], old_access_keys_to_remove = []) {
-        const account_config_path = this.get_account_path_by_name(account_name);
-        await native_fs_utils.update_config_file(this.fs_context, this.accounts_dir_path,
-            account_config_path, JSON.stringify(account_new_data));
-
-        if (new_access_keys_to_link.length > 0) {
-            for (const access_keys of new_access_keys_to_link) {
-                const new_access_key_path = this.get_account_path_by_access_key(access_keys.access_key);
-                const account_config_relative_path = this.get_account_relative_path_by_name(account_name);
-                await nb_native().fs.symlink(this.fs_context, account_config_relative_path, new_access_key_path);
-            }
-        }
-
-        if (old_access_keys_to_remove.length > 0) {
-            for (const access_key of old_access_keys_to_remove) {
-                const cur_access_key_path = this.get_account_path_by_access_key(access_key);
-                await nb_native().fs.unlink(this.fs_context, cur_access_key_path);
-            }
+    async update_account_config_file(account_new_data, options = {}) {
+        const { name, _id, owner = undefined } = account_new_data;
+        const data_string = JSON.stringify(account_new_data);
+        const account_path = this.get_identity_path_by_id(_id);
+        const account_dir_path = this.get_identity_dir_path_by_id(_id);
+        await native_fs_utils.update_config_file(this.fs_context, account_dir_path, account_path, data_string);
+
+        if (options.old_name) {
```

Comment:
old_name? in the CLI we used curr_name - both of them sound confusing.

Reply:
do you have a better suggestion?

Reply:
No, leave as is, thanks.